editoast/src/views/work_schedules.rs

```diff
@@ -1,44 +1,66 @@
+use crate::core::pathfinding::TrackRange as CoreTrackRange;
+use crate::error::InternalError;
+use crate::error::Result;
+use crate::models::prelude::*;
+use crate::models::work_schedules::WorkSchedule;
+use crate::models::work_schedules::WorkScheduleGroup;
+use crate::models::work_schedules::WorkScheduleType;
+use crate::views::path::projection::Intersection;
+use crate::views::path::projection::PathProjection;
+use crate::views::AuthenticationExt;
+use crate::views::AuthorizationError;
+use crate::AppState;
 use axum::extract::Json;
+use axum::extract::Path;
 use axum::extract::State;
+use axum::response::IntoResponse;
 use axum::Extension;
 use chrono::DateTime;
 use chrono::Utc;
 use derivative::Derivative;
 use editoast_authz::BuiltinRole;
 use editoast_derive::EditoastError;
 use editoast_models::DbConnectionPoolV2;
+use editoast_schemas::infra::{Direction, TrackRange};
 use serde::de::Error as SerdeError;
 use serde::Deserialize;
 use serde::Serialize;
 use std::result::Result as StdResult;
 use thiserror::Error;
-use utoipa::ToSchema;
-
-use crate::core::pathfinding::TrackRange as CoreTrackRange;
-use crate::error::InternalError;
-use crate::error::Result;
-use crate::models::prelude::*;
-use crate::models::work_schedules::WorkSchedule;
-use crate::models::work_schedules::WorkScheduleGroup;
-use crate::models::work_schedules::WorkScheduleType;
-use crate::views::path::projection::Intersection;
-use crate::views::path::projection::PathProjection;
-use crate::views::AuthenticationExt;
-use crate::views::AuthorizationError;
-use crate::AppState;
-use editoast_schemas::infra::{Direction, TrackRange};
+use utoipa::{IntoParams, ToSchema};
+use uuid::Uuid;
 
 crate::routes! {
     "/work_schedules" => {
         create,
         "/project_path" => project_path,
+        "/group" => {
+            create_group,
+            list_groups,
+            "/{id}" => {
+                delete_group,
+                get_group,
+                put_in_group,
+            },
+        },
     },
 }
 
 editoast_common::schemas! {
+    WorkSchedule,
     WorkScheduleCreateForm,
     WorkScheduleCreateResponse,
+    WorkScheduleGroupCreateForm,
+    WorkScheduleGroupCreateResponse,
+    WorkScheduleInGroupCreateForm,
     WorkScheduleItemForm,
+    WorkScheduleType,
```

Comment:
I believe most of these schemas are only used once, so they can be inlined if you want.

Reply:
Fixed in 64d4756f07e49aab97df92b70c1bfa4e38107ae6